### PR TITLE
[Improvement-3369][api] Introduce alert group and users service interface for clear code

### LINKED ISSUE #3369: [Improvement][api] Introduce interface about api module service
**Describe the question**
Introduce interface about api module service for clearly service module.

**What are the current deficiencies and the benefits of improvement**
- A clear and concise service will help contributor or user to ready the code and locate the function.

**Which version of DolphinScheduler:**
 -[dev]

**Describe alternatives you've considered**
Introduce service interface

### PR DESCRIPTION
**Pull request by chengshiwen**
## What is the purpose of the pull request

*Introduce alert group and users service interface for clear code and close pr #3417 which has conflicts and not updated for a long time*

Compared to pr #3417 with **20 code smells** half year ago, there are **22 code smells** in this pr.

This closes #3369

## Brief change log

  - *Add AlertGroupServiceImpl and change AlertGroupService to interface*
  - *Add UsersServiceImpl and change UsersService to interface*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

**Comment by CalvinKirs** [comment]
There are still many code smells, can you solve it first?

**Comment by chengshiwen** [comment]
> There are still many code smells, can you solve it first?

I will try to solve it

**Comment by chengshiwen** [comment]
@CalvinKirs Now only 8 code smells

image

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertGroupServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        Date now = new Date();
+
+        if (StringUtils.isNotEmpty(groupName)) {
+            alertGroup.setGroupName(groupName);
+        }
+        alertGroup.setDescription(desc);
+        alertGroup.setUpdateTime(now);
+        alertGroup.setCreateUserId(loginUser.getId());
+        alertGroup.setAlertInstanceIds(alertInstanceIds);
+        // updateProcessInstance
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertGroupServiceImpl.java]: Delete the meaningless comment(// updateProcessInstance)

Comment (chengshiwen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertGroupServiceImpl.java]: Done

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**